app/common/Exceptions.scala

```diff
@@ -40,4 +40,9 @@ trait Exceptions {
         s"Value not found for protection type in $functionName"
       )
 
+  case class RequiredNotFoundProtectionModelException(functionName: String)
+      extends Exception(
+        s"Value not found for protection type in $functionName"
+      )
```

Comment:
Message of this exception looks a bit misleading. 
Additionally, the class name is not clear - I am not able to figure out what it's for just by reading it. It doesn't sound English, tbh. Hopefully, you can come up with a better name?